tap_google_sheets/client.py

```diff
@@ -115,12 +115,18 @@ def raise_for_error(response):
                 # There is nothing we can do here since Google has neither sent
                 # us a 2xx response nor a response content.
                 return
+            # Fetch the status code from the response object itself.
+            status_code = response.status_code
             response = response.json()
             if ('error' in response) or ('errorCode' in response):
-                message = '%s: %s' % (response.get('error', str(error)),
-                                      response.get('message', 'Unknown Error'))
-                error_code = response.get('error', {}).get('code')
```

Comment:
This is what was throwing the error: the `error` is in the response, but the object is a str and fails to `get('code')`